pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -169,9 +176,10 @@ type PriorityQueue struct {
 
 	// activeQLock synchronizes all operations related to activeQ.
 	// It protects activeQ, inFlightPods, inFlightEvents, schedulingCycle and closed fields.
-	// Caution: DO NOT take nominator.lock after taking activeQLock,
-	// you should take nominator.lock first if you need two locks,
-	// otherwise the queue could end up deadlock.
+	// Caution: DO NOT take "lock" after taking "activeQLock".
+	// You should always take "lock" first, otherwise the queue could end up in deadlock.
+	// "nominator.nLock" should not be taken while taking "activeQLock".
+	// Correct locking order is: lock > activeQLock > nominator.nLock.
```

Comment:
```suggestion
	// Correct locking order is: lock > (activeQLock | nominator.nLock).
```
?

Because you say `"nominator.nLock" should not be taken while taking "activeQLock"`

Reply:
The order is correct. Rephrased the comments a bit to be more straightforward